src/chaosinventory/base/serializers.py

```diff
@@ -0,0 +1,143 @@
+from rest_framework import serializers
+
+from .models import (
+    DataType, Entity, Item, Location, LocationData, Product, ProductData, Tag,
+)
+
+
+class BasicLocationSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Location
+        fields = [
+            'id',
+            'name',
+            'note',
+            'in_location'  # TODO: Make recursive?
+        ]
+
+
+class TagSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Tag
+        fields = [
+            'id',
+            'parent'
+        ]
+
+
+class BasicLocationDataSerializer(serializers.ModelSerializer):
+    type = serializers.ReadOnlyField(source='type.name')
+
+    class Meta:
+        model = LocationData
+        fields = [
+            'id',
+            'value',
+            'type'
+        ]
+
+
+class BasicProductDataSerializer(serializers.ModelSerializer):
+    type = serializers.ReadOnlyField(source='type.name')
+
+    class Meta:
+        model = ProductData
+        fields = [
+            'id',
+            'value',
+            'type'
+        ]
+
+
+class DataTypeSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = DataType
+        fields = [
+            'id',
+            'name',
+            'note'
+        ]
+
+
+class LocationDataSerializer(serializers.ModelSerializer):
+    type = serializers.ReadOnlyField(source='type.name')
+    location = BasicLocationSerializer()
+
+    class Meta:
+        model = LocationData
+        fields = [
+            'id',
+            'value',
+            'type',
+            'location'
```

Comment:
As far as I can tell that's a complicated topic. Currently, we are doing some amount of recursion on the obejcts. But if we want to be able to simply attach tags to $Things, it would be easiest to only serve the PK of the Tags as list and not the name, id and the type of it.

Hm but we could also write some custom handlers that when creating objects, the id is sufficient, but give the full output when GETting objects. See https://www.django-rest-framework.org/api-guide/relations/#nested-relationships for reference.